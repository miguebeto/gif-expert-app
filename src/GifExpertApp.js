import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  //Se usan arreglos de esta manera solamente cuando do vamos hacer cambios mediante eventos, nunca debe mutarse
  // const categories = ['One punch', 'Samuray x', 'Dragon ball'];

  //usamos useState cuando la colección será mutada o cambiada luego de cargado el componente
  const [categories, setCategories] = useState(defaultCategories);

  // const handleAdd = ()=> {
  //     //formas de agregar elementos al arreglo en react
  //     // setcategories(categories.concat('HunterxHunter'));
  //     // setcategories([ 'HunterxHunter', ...categories]);
  //     setcategories(categories=> [...categories, 'HunterxHunter']);

  // }

  return (
    <>
      <h2 className={"animate__animated animate__flash"}>GifExpertApp</h2>
      {/* Mandamos la referencia de la funcion seteadora via props */}
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {/* recorremos el arreglo y lo pasamos como props al componente  */}
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
