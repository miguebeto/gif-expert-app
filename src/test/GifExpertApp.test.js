import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en el componente <GifExpertApp />", () => {
  test("debe retorna el componente en su forma original", () => {
    //comprueba que el componente se encuentre sin modificaciones en el renderizado
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar una lista de categorias", () => {
    //creamos una variable con el areglo de las categorias que se van a pasar en las props del componente y en el initialstate
    const categories = ["Evangelion", "Naruto"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    //comprueba que el componente se encuentre sin modificaciones en el renderizado
    expect(wrapper).toMatchSnapshot();

    //comprueba que el componente est√° renderizando el componente <GifGrid /> por cada elemento del arreglo
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
