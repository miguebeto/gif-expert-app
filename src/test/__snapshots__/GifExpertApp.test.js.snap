// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flash"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Evangelion"
      key="Evangelion"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe retorna el componente en su forma original 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flash"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
